RealmTasks Android/app/src/main/java/io/realm/realmtasks/list/TaskAdapter.java

```diff
@@ -41,12 +44,29 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         if (task.isValid()) {
             final TextView text = itemViewHolder.getText();
             text.setText(task.getText());
+
+            Date taskDate = task.getDate();
+            if(taskDate != null) {
+                CharSequence naturalDateString = naturalDateFrom(taskDate);
+                itemViewHolder.setMetadataText(naturalDateString);
+            } else {
+                itemViewHolder.setMetadataText(null);
+            }
             narrowRightMargin(text);
             narrowRightMargin(itemViewHolder.getEditText());
             itemViewHolder.setCompleted(task.isCompleted());
         }
     }
 
+    private String naturalDateFrom(@NonNull Date taskDueDate) {
+        return DateUtils.getRelativeDateTimeString(
+                context,
+                taskDueDate.getTime(),
+                DateUtils.DAY_IN_MILLIS,
+                DateUtils.WEEK_IN_MILLIS, 0).toString()
+                .replace(",", " at");
```

Comment:
Thanks @nhachicha for suggesting.  I had looked at that.  Using `getRelativeTimeSpanString(Context c, long millis,  boolean withPreposition)` yields only the date when not today (i.e. `tomorrow`, `on Feb 28`, `at 7pm`), which farther from what I want.

I think I'll go with what I had, except that I won't replace the comma at the end.